scripts/update_deps.py

```diff
@@ -435,8 +441,13 @@ def CMakeBuild(self):
         # Speed up the build.
         if platform.system() == 'Linux' or platform.system() == 'Darwin':
             cmake_cmd.append('--')
-            cmake_cmd.append('-j{ncpu}'
-                             .format(ncpu=multiprocessing.cpu_count()))
+            num_make_jobs = multiprocessing.cpu_count()
+            if 'MAKE_JOBS' in os.environ:
+                try:
+                    num_make_jobs = min(num_make_jobs, int(os.environ['MAKE_JOBS']))
+                except ValueError:
+                    pass
```

Comment:
Is there a reason we are catching an error only to suppress it?

Reply:
Yes, though I clearly neglected to document it it comments...

I took it as an objective to ensure that these changes would not break any existing environments.  In the (IMHO unlikely, but possible) case that some environment using update_deps happened to have a non-numeric MAKE_JOBS environment variable, I didn't want to either crash or produce an unexpected warning message, so I just continued as the code would have before.

If you disagree with the logic, I'm happy to make a different response - I'm not particularly fond of silent exception handling either.

Reply:
I think it would be good to let the user know that their 'MAKE_JOBS' environment variable is in a weird state and tell them how many jobs we will be performing instead.

This would hopefully help the user debug their system instead of thinking that our script doesn't listen to the environment variables they have set.

Reply:
Done...